tests/rados/test_node_drain_customer_bug.py

```diff
@@ -0,0 +1,292 @@
+    3. Parallely execute the ceph mgr fail command
+    4. Check that exception occurs on the cluster
+    5. Perform the following workaround  steps-
+          5.1 ceph config-key rm mgr/cephadm/osd_remove_queue
+          5.2 ceph mgr fail
+    6. If exception occured then check for the Traceback logs and perform the workaround
+    7. After the verification add the drained node back to the cluster
+    """
+    log.info(run.__doc__)
+    config = kw["config"]
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    rados_obj = RadosOrchestrator(node=cephadm)
+    mgr_obj = MgrWorkflows(node=cephadm)
+    installer = ceph_cluster.get_nodes(role="installer")[0]
+    service_obj = ServiceabilityMethods(cluster=ceph_cluster, **config)
+    ceph_nodes = kw.get("ceph_nodes")
+    config = kw["config"]
+
+    replicated_config = config.get("replicated_pool")
+    pool_name = replicated_config["pool_name"]
+
+    if not rados_obj.create_pool(pool_name=pool_name):
+        log.error("Failed to create the  Pool")
+        return 1
+
+    rados_obj.bench_write(pool_name=pool_name, byte_size="5M", rados_write_duration=90)
+    mgr_daemon = Thread(
+        target=background_mgr_task, kwargs={"mgr_object": mgr_obj}, daemon=True
+    )
+    # Printing the hosts in cluster
+    cmd_host_ls = "ceph orch host ls"
+    out = rados_obj.run_ceph_command(cmd=cmd_host_ls)
+    log.debug(f"The hosts in the cluster before starting the test are - {out}")
+
+    mgr_host_object_list = []
+    for node in ceph_nodes:
+        if node.role == "mgr":
+            mgr_host_object_list.append(node)
+            log.debug(f"The mgr host node is{node.hostname}")
+
+    mgr_daemon_list = mgr_obj.get_mgr_daemon_list()
+    log.debug(f"The MGR daemons list are -{mgr_daemon_list}")
+
+    log_lines = [
+        "mgr load Traceback",
+        "TypeError: __init__() got an unexpected keyword argument 'original_weight'",
+    ]
+
+    ceph_version = rados_obj.run_ceph_command(cmd="ceph version")
+    log.info(f"Current version on the cluster : {ceph_version}")
+    match_str = re.match(
+        r"ceph version (\d+)\.(\d+)\.(\d+)-(\d+)", ceph_version["version"]
+    )
+    major, minor, patch, build = match_str.groups()
+    bug_exists = False
+    exceptional_flag = False
+    if int(major) < 18:
+        bug_exists = True
+    elif int(major) == 18 and int(minor) < 2:
+        bug_exists = True
+    elif int(major) == 18 and int(minor) == 2 and int(patch) < 1:
+        bug_exists = True
+    elif int(major) == 18 and int(minor) == 2 and int(patch) == 1 and int(build) < 234:
+        bug_exists = True
+
+    if bug_exists:
+        log.info(f"The bug exists and  the ceph version is - {ceph_version}")
+    else:
+        log.info(f"The bug not exists and the ceph version is - {ceph_version}")
+
+    # Check the labels
+    cmd_host_ls = "ceph orch host ls"
+    out = rados_obj.run_ceph_command(cmd=cmd_host_ls)
+    drain_host = None
+    for node in out:
```

Comment:
The same node is not being selected for draining process, once the issue was already reproduced before upgrade.

Please check this. 

Before upgrade :
``` 2024-10-14 04:40:57,621 (cephci.test_node_drain_customer_bug) [INFO] - cephci.ceph.ceph.py:1570 - Execute cephadm shell -- sudo ceph osd ls-tree ceph-rados-auto-887u7t-node3 -f json on 10.0.66.155
2024-10-14 04:40:58,734 (cephci.test_node_drain_customer_bug) [INFO] - cephci.ceph.ceph.py:1600 - Execution of cephadm shell -- sudo ceph osd ls-tree ceph-rados-auto-887u7t-node3 -f json on 10.0.66.155 took 1.112087 seconds.
2024-10-14 04:40:58,737 (cephci.test_node_drain_customer_bug) [INFO] - cephci.tests.rados.test_node_drain_customer_bug.py:115 - The OSDs in the drain node before starting the test - [4, 9, 14, 19] 
2024-10-14 04:40:58,738 (cephci.test_node_drain_customer_bug) [INFO] - cephci.ceph.rados.serviceability_workflows.py:231 - Identified host : ceph-rados-auto-887u7t-node3 to be removed from the cluster ```

After upgrade:
```2024-10-14 05:01:33,686 (cephci.test_cephadm_upgrade) [INFO] - cephci.ceph.ceph.py:1570 - Execute cephadm shell -- sudo ceph osd ls-tree ceph-rados-auto-887u7t-node9 -f json on 10.0.66.155
2024-10-14 05:01:34,732 (cephci.test_cephadm_upgrade) [INFO] - cephci.ceph.ceph.py:1600 - Execution of cephadm shell -- sudo ceph osd ls-tree ceph-rados-auto-887u7t-node9 -f json on 10.0.66.155 took 1.044808 seconds.
2024-10-14 05:01:34,733 (cephci.test_cephadm_upgrade) [INFO] - cephci.tests.rados.test_node_drain_customer_bug.py:115 - The OSDs in the drain node before starting the test - [0, 5, 10, 15] 
2024-10-14 05:01:34,734 (cephci.test_cephadm_upgrade) [INFO] - cephci.ceph.rados.serviceability_workflows.py:231 - Identified host : ceph-rados-auto-887u7t-node9 to be removed from the cluster ```

Reply:
I included one more check to identify the correct node.